receiver/hostmetricsreceiver/internal/scraper/processscraper/process_filter.go

```diff
@@ -0,0 +1,141 @@
+func (p *processFilter) includePid(pid int32) bool {
+	return (len(p.includePidFilter) == 0 || findInt(p.includePidFilter, pid)) &&
+		(len(p.excludePidFilter) == 0 || !findInt(p.excludePidFilter, pid))
+}
+
+// findInt searches an int map for a specific integer.  If the
+// int is found in the map return true.  Otherwise return false
+func findInt(intMap map[int32]struct{}, intMatch int32) bool {
+	_, ok := intMap[intMatch]
+	return ok
+}
+
+// createFilter receives a filter config and createa a processFilter based on the config settings
+func createFilter(filterConfig FilterConfig) (*processFilter, error) {
+	var err error
+	filter := processFilter{}
+
+	filter.includeExecutableNameFilter, err = filterhelper.NewIncludeFilterHelper(filterConfig.IncludeExecutableNames.ExecutableNames, &filterConfig.IncludeExecutableNames.Config, "executable_name")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.excludeExecutableNameFilter, err = filterhelper.NewExcludeFilterHelper(filterConfig.ExcludeExecutableNames.ExecutableNames, &filterConfig.ExcludeExecutableNames.Config, "executable_name")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.includeExecutablePathFilter, err = filterhelper.NewIncludeFilterHelper(filterConfig.IncludeExecutablePaths.ExecutablePaths, &filterConfig.IncludeExecutablePaths.Config, "executable_path")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.excludeExecutablePathFilter, err = filterhelper.NewExcludeFilterHelper(filterConfig.ExcludeExecutablePaths.ExecutablePaths, &filterConfig.ExcludeExecutablePaths.Config, "executable_path")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.includeCommandFilter, err = filterhelper.NewIncludeFilterHelper(filterConfig.IncludeCommands.Commands, &filterConfig.IncludeCommands.Config, "commands")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.excludeCommandFilter, err = filterhelper.NewExcludeFilterHelper(filterConfig.ExcludeCommands.Commands, &filterConfig.ExcludeCommands.Config, "commands")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.includeCommandLineFilter, err = filterhelper.NewIncludeFilterHelper(filterConfig.IncludeCommandLines.CommandLines, &filterConfig.IncludeCommandLines.Config, "command_lines")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.excludeCommandLineFilter, err = filterhelper.NewExcludeFilterHelper(filterConfig.ExcludeCommandLines.CommandLines, &filterConfig.ExcludeCommandLines.Config, "command_lines")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.includeOwnerFilter, err = filterhelper.NewIncludeFilterHelper(filterConfig.IncludeOwners.Owners, &filterConfig.IncludeOwners.Config, "owners")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.excludeOwnerFilter, err = filterhelper.NewExcludeFilterHelper(filterConfig.ExcludeOwners.Owners, &filterConfig.ExcludeOwners.Config, "owners")
+	if err != nil {
+		return nil, err
+	}
+
+	filter.includePidFilter = make(map[int32]struct{})
```

Comment:
Regarding the map, I was debating this as well.  In most cases the list will be very short and the memory impact of a linear search will be smaller.  I will update this logic.  

Regarding the use case, in the future when we have the infrastructure to make config changes without an otel collector restart, this could be useful.  Imagine a scenario where a single instance of an executable is consuming large amounts of resources but you don't want to monitor all instances of the process.  You could conceivably update the config without a restart of the collector to monitor just that one process to make sure the situation does not get out of hand.